### PR TITLE
Adding initial types for redux-act

### PR DESCRIPTION
**Pull request by a-xin**
Adding flow types and tests for the two commonly used functions in `redux-act`:
- `createAction`
- `createReducer`

### CODE REVIEW

**Comment by GAntoine** [comment]
Hey @a-xin, I'd like to get this merged in, but since it's been a while, is there anything that needs to be changes or improved?

@h-michael Could I get a hand reviewing this?

File: definitions/npm/redux-act_v1.x.x/flow_v0.34.x-/redux-act_v1.x.x.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+declare module 'redux-act' {
+  declare module.exports: {
+    createReducer<ReducerState>(
+      handlers: {
+        [key: string]: (ReducerState, payload: mixed) => ReducerState,
+      },
```
Comment (h-michael) [inline definitions/npm/redux-act_v1.x.x/flow_v0.34.x-/redux-act_v1.x.x.js]: handlers may be Object or Function.
https://github.com/pauldijou/redux-act/blob/master/src/createReducer.js#L49

File: definitions/npm/redux-act_v1.x.x/flow_v0.34.x-/redux-act_v1.x.x.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      defaultState?: ReducerState,
+    ): {
+      (): (state: ReducerState, payload?: mixed, meta?: mixed) => ReducerState,
+
+      // TODO: Document parameters & add tests
+      options: () => mixed,
+      has: (actionCreator: mixed) => boolean,
+      on: (actionCreator: mixed, reduceFunction: () => mixed) => void,
+      off: (actionCreator: mixed) => void,
```
Comment (h-michael) [inline definitions/npm/redux-act_v1.x.x/flow_v0.34.x-/redux-act_v1.x.x.js]: on and off return 'reducer'
https://github.com/pauldijou/redux-act/blob/master/src/createReducer.js#L32

**Comment by GAntoine** [comment]
@a-xin Are you still interested in applying michael's review? It'd be great to get this merged in, as we don't yet have a definition for `redux-act`.

**Comment by a-xin** [comment]
@GAntoine I'm currently not working on this but happy to take a look at the suggestions (thanks @h-michael). Anyways, I'll only cover the basic cases, if you want to jump in and increase coverage you are more than welcome

**Pull request merged**